src/components/PopupWithForm.js

```diff
@@ -33,14 +32,15 @@ export default class PopupWithForm extends Popup {
   setEventListeners() {
     this._popupElement.addEventListener("submit", () => {
       this._handleFormSubmit(this._getInputValues());
+      this._popupForm.reset();
 
       this.close();
```

Comment:
`      this.close();
`

Please, delete it

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

Reply:
Hi, when I delete line 37, it prevented all forms from closing after clicking the submit button. I tried to get guidance from a tutoring session and we could not figure out how to get them to close without this.close(); in PopupWithForm.js. Can you provide guidance on how this can be achieved?